crates/dojo/bindgen/src/plugins/typescript/generator/enum.rs

```diff
@@ -21,33 +22,80 @@ impl TsEnumGenerator {
             }
         }
     }
-}
 
-impl BindgenModelGenerator for TsEnumGenerator {
-    fn generate(&self, token: &Composite, buffer: &mut Buffer) -> BindgenResult<String> {
-        if token.r#type != CompositeType::Enum || token.inners.is_empty() {
-            return Ok(String::new());
-        }
+    fn generate_simple_enum(
+        &self,
+        token: &Composite,
+        buffer: &mut Buffer,
+    ) -> BindgenResult<String> {
+        Ok(format!(
+            "// Type definition for `{path}` enum
+export const {camel_name} = [
+{variants}
+] as const;
+export type {name} = {{ [key in typeof {camel_name}[number]]: string }};
+export type {name}Enum = CairoCustomEnum;
+",
+            path = token.type_path,
+            name = token.type_name(),
+            camel_name = token.type_name().to_case(Case::Camel),
+            variants = token
+                .inners
+                .iter()
+                .map(|inner| { format!("\t'{}',", inner.name) })
+                .collect::<Vec<String>>()
+                .join("\n")
+        ))
+    }
 
-        self.check_import(token, buffer);
-        let gen = format!(
+    fn generate_custom_enum(
+        &self,
+        token: &Composite,
+        buffer: &mut Buffer,
```

Comment:
Same here, `buffer` is not used anymore, can be removed if not planned to be used (or not part of a trait that must be implemented).